src/Microsoft.Health.Fhir.Liquid.Converter.Tool.UnitTests/ConverterLogicHandlerTest.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.IO;
+using Microsoft.Health.Fhir.Liquid.Converter.Tool;
+using Microsoft.Health.Fhir.Liquid.Converter.Tool.Models;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Tool.UnitTests;
+
+public class ConverterLogicHandlerTest
+{
+    [Fact]
+    public void ConvertWithoutSaving_ShouldThrowDirectoryNotFoundException_WhenTemplateDirectoryDoesNotExist()
+    {
+        var options = new ConverterOptions()
+        {
+            TemplateDirectory = "nonexistent_directory",
+            RootTemplate = "rootTemplate",
+            InputDataContent = "inputDataContent",
+        };
+
+        var exception = Assert.Throws<DirectoryNotFoundException>(() =>
+            ConverterLogicHandler.ConvertWithoutSaving(options));
+
+        Assert.Equal($"Could not find template directory: {options.TemplateDirectory}", exception.Message);
+    }
+
+    [Fact]
+    public void ConvertWithoutSaving_ShouldThrowFileNotFoundException_WhenMetadataFileDoesNotExist()
+    {
+        var options = new ConverterOptions()
+        {
+            TemplateDirectory = "test_directory",
+            RootTemplate = "rootTemplate",
+            InputDataContent = "inputDataContent",
+        };
+
+        Directory.CreateDirectory(options.TemplateDirectory);
+
+        try
+        {
+            var exception = Assert.Throws<FileNotFoundException>(() =>
+                ConverterLogicHandler.ConvertWithoutSaving(options));
+
+            Assert.Equal($"Could not find metadata.json in template directory: {options.TemplateDirectory}.", exception.Message);
+        }
+        finally
+        {
+            Directory.Delete(options.TemplateDirectory);
+        }
+    }
+
+    [Fact]
+    public void ConvertWithoutSaving_ShouldThrowNotImplementedException_WhenDataTypeIsNotSupported()
+    {
+        var options = new ConverterOptions()
+        {
+            TemplateDirectory = "test_directory",
```

Comment:
does dotnet have any built in capability to create a temp directory that we could use instead of hardcoding one (just to ensure it really is a temporary deletable directory - or will `CreateDirectory` error if the directory already exists?)